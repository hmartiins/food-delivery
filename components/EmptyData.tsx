import { Image, ImageSourcePropType, Text, View } from 'react-native';

import { images } from '@/constants';

export const EmptyData = ({
  title,
  subtitle,
  image,
}: {
  title: string;
  subtitle: string;
  image: ImageSourcePropType;
}) => {
  return (
    <View className="flex flex-col items-center justify-center">
      <View className="relative">
        <Image source={images.emptyState} className="mb-8" />

        <View className="absolute inset-0 top-1/2 flex items-center justify-center">
          <View className="flex size-14 items-center justify-center rounded-full bg-primary">
            <Image
              source={image}
              className="size-5 text-white"
              tintColor={'#FFFF'}
              testID="custom-icon"
            />
          </View>
        </View>
      </View>
      <Text
        className="base-bold my-3 text-center text-dark-100"
        testID="empty-data-title"
      >
        {title}
      </Text>
      <Text className="body-regular text-gray-200" testID="empty-data-subtitle">
        {subtitle}
      </Text>
    </View>
  );
};
