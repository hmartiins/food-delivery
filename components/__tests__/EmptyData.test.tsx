import React from 'react';

import { render } from '@testing-library/react-native';

import { images } from '@/constants';

import { EmptyData } from '../EmptyData';

describe('<EmptyData />', () => {
  const defaultProps = {
    title: 'Nothing matched your search',
    subtitle: 'Try a different search term or check for typos.',
    image: images.search,
  };

  it('should render without errors', () => {
    const { getByTestId } = render(<EmptyData {...defaultProps} />);

    const customIcon = getByTestId('custom-icon');
    const emptyDataTitle = getByTestId('empty-data-title');
    const emptyDataSubtitle = getByTestId('empty-data-subtitle');

    expect(emptyDataTitle).toHaveTextContent(defaultProps.title);
    expect(emptyDataSubtitle).toHaveTextContent(defaultProps.subtitle);
    expect(customIcon.props.source).toBe(defaultProps.image);
  });

  describe('snapshots', () => {
    it('should match snapshot with default props', () => {
      const component = render(<EmptyData {...defaultProps} />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
