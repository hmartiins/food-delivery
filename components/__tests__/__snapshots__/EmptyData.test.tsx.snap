// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyData /> snapshots should match snapshot with default props 1`] = `
<View
  className="flex flex-col items-center justify-center"
>
  <View
    className="relative"
  >
    <Image
      className="mb-8"
      source={
        {
          "testUri": "../../../assets/images/empty-state.png",
        }
      }
    />
    <View
      className="absolute inset-0 top-1/2 flex items-center justify-center"
    >
      <View
        className="flex size-14 items-center justify-center rounded-full bg-primary"
      >
        <Image
          className="size-5 text-white"
          source={
            {
              "testUri": "../../../assets/icons/search.png",
            }
          }
          testID="custom-icon"
          tintColor="#FFFF"
        />
      </View>
    </View>
  </View>
  <Text
    className="base-bold my-3 text-center text-dark-100"
    testID="empty-data-title"
  >
    Nothing matched your search
  </Text>
  <Text
    className="body-regular text-gray-200"
    testID="empty-data-subtitle"
  >
    Try a different search term or check for typos.
  </Text>
</View>
`;
